glean_parser/data_review.py

```diff
@@ -0,0 +1,147 @@
+
+    if util.report_validation_errors(all_objects):
+        return 1
+
+    # I tried [\W\Z] but it complained. So `|` it is.
+    reobj = re.compile(f"\\W{bug}\\W|\\W{bug}$")
+    durations = set()
+    metrics_table = ""
+    for category_name, metrics in all_objects.value.items():
+        for metric in metrics.values():
+            if not any([len(reobj.findall(bug)) == 1 for bug in metric.bugs]):
+                continue
+
+            metric_name = util.snake_case(metric.name)
+            category_name = util.snake_case(category_name)
+            one_line_desc = metric.description.replace("\n", " ")
+            sensitivity = ", ".join([s.name for s in metric.data_sensitivity])
+            last_bug = metric.bugs[-1]
+            metrics_table += f"{category_name}.{metric_name} | "
+            metrics_table += f"{one_line_desc} | {sensitivity} | {last_bug}\n"
+
+            durations.add(metric.expires)
+
+    if len(durations) == 1:
+        duration = next(iter(durations))
+        if duration == "never":
+            collection_duration = "This collection will be collected permanently."
+        else:
+            collection_duration = f"This collection has expiry '{duration}'"
+    else:
+        collection_duration = "Parts of this collection expire at different times: "
+        collection_duration += f"{durations}"
+
+    if "never" in durations:
+        collection_duration += "\n<TODO: identify at least one individual here> "
+        collection_duration += "will be responsible for the permanent collections."
+
+    print(
+        """
+!! Reminder: it is your responsibility to complete and check the correctness of
+!! this automatically-generated request skeleton before requesting Data
+!! Collection Review. See https://wiki.mozilla.org/Data_Collection for details.
+
+DATA REVIEW REQUEST
+1. What questions will you answer with this data?
+
+TODO: Fill this in.
+
+2. Why does Mozilla need to answer these questions? Are there benefits for users?
+   Do we need this information to address product or business requirements?
+
+TODO: Fill this in.
+
+3. What alternative methods did you consider to answer these questions?
+   Why were they not sufficient?
+
+TODO: Fill this in.
+
+4. Can current instrumentation answer these questions?
+
+TODO: Fill this in.
+
+5. List all proposed measurements and indicate the category of data collection for each
+   measurement, using the Firefox data collection categories found on the Mozilla wiki.
+
+Measurement Name | Measurement Description | Data Collection Category | Tracking Bug
+---------------- | ----------------------- | ------------------------ | ------------"""
+    )
+    print(metrics_table)
+    print(
+        """
+6. Please provide a link to the documentation for this data collection which
+   describes the ultimate data set in a public, complete, and accurate way.
+
+This collection is Glean so is documented
+[in the Glean Dictionary](https://dictionary.protosaur.dev).
```

Comment:
```suggestion
[in the Glean Dictionary](https://dictionary.telemetry.mozilla.org/).
```

Stable production URL